workers/data_refinery_workers/processors/salmon.py

```diff
@@ -0,0 +1,258 @@
+
+    Also adds the keys "input_file_path" and "output_directory" to
+    job_context so everything is prepared for processing. If the reads
+    are paired then there will also be an "input_file_path_2" key
+    added to job_context for the second read.
+    """
+    # Salmon processor jobs have only one batch per job, but may have
+    # up to two files per batch.
+    batch = job_context["batches"][0]
+    files = batch.files
+
+    for file in files:
+        try:
+            file.download_raw_file(job_context["job_dir_prefix"])
+        except Exception:
+            logger.exception("Exception caught while retrieving raw file %s",
+                             file.get_raw_path(),
+                             processor_job=job_context["job_id"],
+                             batch=batch.id)
+
+            failure_template = "Exception caught while retrieving raw file {}"
+            job_context["job"].failure_reason = failure_template.format(file.name)
+            job_context["success"] = False
+            return job_context
+
+    job_context["input_file_path"] = files[0].get_temp_pre_path(job_context["job_dir_prefix"])
+    # Salmon outputs an entire directory of files, so create a temp
+    # directory to output it to until we can zip it to
+    # files[0].get_temp_post_path()
+    job_context["output_directory"] = os.path.join(
+        files[0].get_temp_dir(job_context["job_dir_prefix"]), "output")
+    os.makedirs(job_context["output_directory"], exist_ok=True)
+
+    if len(files) == 2:
+        job_context["input_file_path_2"] = files[1].get_temp_pre_path(
+            job_context["job_dir_prefix"])
+
+    return job_context
+
+
+def _determine_index_length(job_context: Dict) -> Dict:
+    """Determines whether to use the long or short salmon index.
+
+    Adds the key 'kmer_size' to the job_context with a value of '23'
+    if the short index is appropriate or '31' if the long index is
+    appropriate. For more information on index length see the
+    _create_index function of the transcriptome_index processor.
+    """
+    total_base_pairs = 0
+    number_of_reads = 0
+    counter = 1
+    with gzip.open(job_context["input_file_path"], "rt") as input_file:
+        for line in input_file:
+            if counter % 4 == 2:
+                total_base_pairs += len(line.replace("\n", ""))
+                number_of_reads += 1
+
+            counter += 1
+
+    if "input_file_path_2" in job_context:
+        with gzip.open(job_context["input_file_path_2"], "rt") as input_file:
+            for line in input_file:
+                if counter % 4 == 2:
+                    total_base_pairs += len(line.replace("\n", ""))
+                    number_of_reads += 1
+
+                counter += 1
+
+    if total_base_pairs / number_of_reads > 75:
+        job_context["kmer_size"] = "31"
+    else:
+        job_context["kmer_size"] = "23"
+
+    return job_context
+
+
+def _download_index(job_context: Dict) -> Dict:
```

Comment:
A docstring for this method would be helpful

Reply:
Very true, good call.